### PR TITLE
tests: basic functional tests for CoinJoin

### PR DESCRIPTION
**Pull request by Munkybooty**
Going to be opening this as a draft because i am not finished but it will probably be better to have more eyes on it than me continuously roadblocking myself at the moment

### CODE REVIEW

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
no need for `nVersion < MIN_PEER_PROTO_VERSION` check, such peers are disconnected right away and should never get this far

File: src/coinjoin/client.cpp [hunk 1]
```diff
@@ -142,6 +144,7 @@ void CCoinJoinClientSession::ProcessMessage(CNode* pfrom, const std::string& msg
     if (!masternodeSync.IsBlockchainSynced()) return;
 
     if (msg_type == NetMsgType::DSSTATUSUPDATE) {
+
```
Comment (UdjinM6) [inline src/coinjoin/client.cpp:147]:
```suggestion
```

File: src/coinjoin/client.cpp [hunk 2]
```diff
@@ -153,6 +156,7 @@ void CCoinJoinClientSession::ProcessMessage(CNode* pfrom, const std::string& msg
         ProcessPoolStateUpdate(psssup);
 
     } else if (msg_type == NetMsgType::DSFINALTX) {
+
```
Comment (UdjinM6) [inline src/coinjoin/client.cpp:159]:
```suggestion
```

File: src/coinjoin/client.cpp [hunk 3]
```diff
@@ -173,6 +177,7 @@ void CCoinJoinClientSession::ProcessMessage(CNode* pfrom, const std::string& msg
         SignFinalTransaction(txNew, pfrom, connman);
 
     } else if (msg_type == NetMsgType::DSCOMPLETE) {
+
```
Comment (UdjinM6) [inline src/coinjoin/client.cpp:180]:
```suggestion
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
👍 
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**